tests/buffer/buffer_storage_common.h

```diff
@@ -39,53 +41,84 @@ template <typename alloc, typename T, int size, int dims>
 class buffer_storage_test {
  public:
   void operator()(util::logger &log, cl::sycl::range<dims> r) {
-    std::unique_ptr<T[]> data(new T[size]);
-    std::shared_ptr<T> data_shrd(new T[size], [](T *data) { delete[] data; });
+    auto deleter = [](T *data) { delete[] data; };
+
+    // Case 1 - Raw pointer
+    auto data_final1 = std::make_unique<T[]>(size);
+
+    // Case 2 - Null pointer
+    T *data_final2 = nullptr;
+
+    // Case 3 - Weak pointer
+    std::shared_ptr<T> data_shared_ptr(new T[size], deleter);
+    std::weak_ptr<T> data_final3 = data_shared_ptr;
+
+    // Case 4 - Shared pointer
+    std::shared_ptr<T> data_final4(new T[size], deleter);
+
+    // Case 5 - Vector data
     std::vector<T> data_vector;
     data_vector.reserve(size);
+    auto data_final5 = data_vector.begin();
+
+    check_write_back(log, r, data_final1.get());
+    check_write_back(log, r, data_final2, true);
```

Comment:
```suggestion
    check_write_back(log, r, data_final2, true /*is_nullptr*/);
```

Or something like that

Reply:
Done.